### DISCUSSION TITLE
Update staking logic: increase validator threshold

### DISCUSSION THREAD

**Discussion by devsMynting** [0xPolygon/polygon-edge#556] [Q&A]
Hey 👋, 

I'm trying to figure out how one can update the staking contract logic (such as updating the validator threshold). I've tried to update the bytecode & storage variables post genesis, and also fork the repo to update helper/staking.go & corresponding dependencies; however, in all cases the chain deploys, but the staking contract becomes unresponsive (all calls return an RPC error).

Is there some way to update the staking logic that I may have missed — would appreciate any documentation or help. Thanks!

**Comment by samisbakedham** [comment]
I have the same question 

> Hey 👋, 
> 
> 
> 
> I'm trying to figure out how one can update the staking contract logic (such as updating the validator threshold). I've tried to update the bytecode & storage variables post genesis, and also fork the repo to update helper/staking.go & corresponding dependencies; however, in all cases the chain deploys, but the staking contract becomes unresponsive (all calls return an RPC error).
> 
> 
> 
> Is there some way to update the staking logic that I may have missed — would appreciate any documentation or help. Thanks!

**Comment by laviniat1996** [comment]
Hi @devsMynting ,

There is no way to update the validator threshold once the contract has been deployed. After you have started the POS network, you cannot update things like minimum/maximum validator number or threshold for now. You can add a new validator or remove one (stake/unstake).
